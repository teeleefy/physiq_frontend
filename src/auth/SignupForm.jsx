import { Form, FormGroup, Input, Button, Label, Alert } from "reactstrap";
import {useState} from 'react'
import { useNavigate } from "react-router-dom";
import './styles/Signup.css'
import { toast } from 'react-toastify';

function SignupForm({signup}){
    const INITIAL_STATE = { email:"", password: "", name: ""};
    const [formData, setFormData] = useState(INITIAL_STATE);
    const [formErrors, setFormErrors] = useState([]);
    const navigate = useNavigate();
    
    async function handleSubmit(evt){
        evt.preventDefault();
        // console.log(formData);
        let result = await signup(formData);
        if(result.success){
            toast.success('Registation Successful!', {
                position: "top-center",
                autoClose: 2000,
                hideProgressBar: false,
                closeOnClick: true,
                pauseOnHover: true,
                draggable: true,
                progress: undefined,
                theme: "dark",
                });
            setFormData(INITIAL_STATE);
            navigate('/');
        } else{
            setFormErrors(result.errors);
        }
    };

    const handleChange = evt => {
        const { name, value }= evt.target;
        setFormData(fData => ({
          ...fData,
          [name]: value
        }));
      };

    return(
        <>
            
            <Form className="Signup-Form" onSubmit={handleSubmit}>
            <h1 className="Signup-h1">Sign Up</h1>
            <FormGroup>
                <Label for="email">
                    <b>Email</b>
                </Label>
                <Input
                id="email"
                name="email"
                value={formData.email}
                minLength={6}
                maxLength={60}
                placeholder="Enter email..."
                type="email"
                onChange={ handleChange }
                />
                </FormGroup>
                <FormGroup>
                <Label for="password">
                    <b>Password</b>
                </Label>
                <Input
                id="password"
                name="password"
                placeholder="Enter password..."
                value={formData.password}
                minLength={8}
                type="password"
                onChange={ handleChange }
                />
                </FormGroup>
                <FormGroup>
                <Label for="name" >
                    <b>Family Name</b>
                </Label>
                <Input
                id="name"
                name="name"
                value={formData.name}
                placeholder="Example: The Smith Family"
                minLength={1}
                maxLength={75}
                type="text"
                onChange={ handleChange }
                />
            </FormGroup>
            
            

                {formErrors.length
                    ? formErrors.map(error => <Alert color="danger">{error}</Alert>)
                    : null
                }

                <Button className="btn" type="submit">Submit</Button>
            </Form>
        </>
     )
}

export default SignupForm;