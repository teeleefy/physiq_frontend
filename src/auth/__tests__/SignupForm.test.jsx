import { renderWithRouter, screen } from '../../../__tests__/test-tools';
import SignupForm from '../SignUpForm.jsx';

describe('SignupForm', () => {
  it('renders the LoginForm component', () => {
    renderWithRouter(<SignupForm />)
    
    screen.debug(); // prints out the jsx in the LoginForm component unto the command line
  })
})

it("matches snapshot", function() {
  const { asFragment } = renderWithRouter(<SignupForm />);
  expect(asFragment()).toMatchSnapshot();
});