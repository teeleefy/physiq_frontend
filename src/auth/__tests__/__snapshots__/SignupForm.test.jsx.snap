// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="Signup-Form"
  >
    <h1
      class="Signup-h1"
    >
      Sign Up
    </h1>
    <div
      class="mb-3"
    >
      <label
        class="form-label"
        for="email"
      >
        <b>
          Email
        </b>
      </label>
      <input
        class="form-control"
        id="email"
        maxlength="60"
        minlength="6"
        name="email"
        placeholder="Enter email..."
        type="email"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="form-label"
        for="password"
      >
        <b>
          Password
        </b>
      </label>
      <input
        class="form-control"
        id="password"
        minlength="8"
        name="password"
        placeholder="Enter password..."
        type="password"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="form-label"
        for="name"
      >
        <b>
          Family Name
        </b>
      </label>
      <input
        class="form-control"
        id="name"
        maxlength="75"
        minlength="1"
        name="name"
        placeholder="Example: The Smith Family"
        type="text"
        value=""
      />
    </div>
    <button
      class="btn btn btn-secondary"
      type="submit"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
